ocean_lib/models/test/test_erc721_token.py

```diff
@@ -641,3 +644,531 @@ def test_fail_transfer_function(web3, config, publisher_wallet, consumer_wallet)
         == "execution reverted: VM Exception while processing transaction: revert ERC721: transfer caller is not "
         "owner nor approved"
     )
+
+
+def test_transfer_nft(web3, config, publisher_wallet, consumer_wallet, factory_router):
```

Comment:
This is too long. Break it into 3+ test methods please:)

Reply:
I will refactor in #758 and specify the modifications there. Thank you!